cmd/pebble/cmd_exec.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2021 Canonical Ltd
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 3 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package main
+
+import (
+	"errors"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/jessevdk/go-flags"
+	"golang.org/x/sys/unix"
+
+	"github.com/canonical/pebble/client"
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/canonical/pebble/internal/ptyutil"
+)
+
+type cmdExec struct {
+	clientMixin
+	WorkingDir string        `long:"cwd"`
+	Env        []string      `long:"env"`
+	User       string        `long:"user"`
+	Group      string        `long:"group"`
+	Timeout    time.Duration `long:"timeout"`
+	Terminal   bool          `short:"t" long:"terminal"`
+	NoTerminal bool          `short:"T" long:"no-terminal"`
+	Positional struct {
+		Command string `positional-arg-name:"<command>" required:"1"`
+	} `positional-args:"yes"`
```

Comment:
Docker docks aren't very expansive on this subject, but from what I can tell from testing, without `-i` the `docker exec` and `docker run` commands don't hook up stdin at all. I don't really know how that's useful -- I always use `-it` with Docker as well, and it's in all the example! So maybe they just got the default wrong.

Both `lxc exec` and `ssh` reverse this (the default is to connect stdin). That's a better default for `pebble exec` as well, I think. Those programs have a `-n` flag to disable stdin (I'm not actually sure when that would be needed, though in the SSH docs it says "This must be used when ssh is run in the background"). I don't understand the need for `-n`, but it will be easy to add later if we do need it.

Reply:
Agreed on all counts. Marking this as resolved.